api/user/resolvers.ts

```diff
@@ -48,58 +54,103 @@ export class Mutation {
     _root: Record<string, never>,
     { email, password }: MutationSignupArgs,
     context: Context
-  ): Promise<User> {
+  ): Promise<SignupPayload> {
     const newUser = await this.authService.createAccount(email, password)
-    await context.login(newUser)
+    if ('user' in newUser) void context.login(await newUser.user)
```

Comment:
Please rename the variable to `newUserPayload` (since it's no longer only the new user)